script/Connector-Axelar.s.sol

```diff
@@ -17,17 +18,21 @@ contract ConnectorAxelarScript is Script {
     function run() public {
         vm.startBroadcast();
 
-        address escrow_ = address(new ConnectorEscrow{ salt: SALT }());
         address tokenFactory_ = address(new RestrictedTokenFactory{ salt: SALT }());
         address memberlistFactory_ = address(new MemberlistFactory{ salt: SALT }());
         CentrifugeConnector connector =
-            new CentrifugeConnector{ salt: SALT }(escrow_, tokenFactory_, memberlistFactory_);
+            new CentrifugeConnector{ salt: SALT }(tokenFactory_, memberlistFactory_);
+        address escrow_ = address(new ConnectorEscrow{ salt: SALT }());
+        connector.file("escrow", escrow_);
 
         ConnectorAxelarRouter router = new ConnectorAxelarRouter{ salt: SALT }(
                 address(connector),
                 address(vm.envAddress("AXELAR_GATEWAY"))
         );
         connector.file("router", address(router));
+        ConnectorGateway gateway = new ConnectorGateway{ salt: SALT }(address(connector), address(router));
+        // TODO: We should be able to make the gateway address immutable on routers once we're deploying deterministically.
```

Comment:
This is not really true. Deploying deterministically means deploying the same byte code on the same address. If we deploy a new gateway contract, it will have a different byte code, and so the address will be different.

The purpose of deterministic deployments instead is to deploy the same bytecode (the token contract) to the same address on different chains.

Reply:
Ooh I thought you could "predict" what the address of a deterministically deployed contract would be?